File: shared/types/student.ts
export interface Student {
  id: string;
  first_name: string;
  last_name: string;
  age?: number;
  birth_date?: string;
  parent_contact?: string;
  phone?: string;
  telegram?: string;
  course_ids: string[];
  start_date?: string;
  photo_url?: string;
  is_active: boolean;
  created_at: string;
}

File: backend/app/models/student.py
from pydantic import BaseModel
from typing import Optional


class Student(BaseModel):
    id: str
    first_name: str
    last_name: str
    age: Optional[int] = None
    birth_date: Optional[str] = None
    parent_contact: Optional[str] = None
    parent_name: Optional[str] = None
    parent_relation: Optional[str] = None
    phone: Optional[str] = None
    telegram: Optional[str] = None
    course_ids: list[str] = []
    start_date: Optional[str] = None
    photo_url: Optional[str] = None
    is_active: bool = True
    created_at: str = ""
    user_id: str = ""
    updated_at: str = ""
